plotly/offline/offline.py

```diff
@@ -204,11 +213,30 @@ def _plot_html(figure_or_data, show_link, link_text, validate,
         link_text = link_text.replace('plot.ly', link_domain)
         config['linkText'] = link_text
 
-    script = 'Plotly.newPlot("{id}", {data}, {layout}, {config})'.format(
-        id=plotdivid,
-        data=jdata,
-        layout=jlayout,
-        config=jconfig)
+    if 'frames' in figure_or_data:
+        script = '''
+        Plotly.plot(
+            '{id}',
+            {data},
+            {layout},
+            {config}
+        ).then(function () {add_frames}).then(function(){animate})
```

Comment:
Yeah, should just be a simplification. Long story short, given `data`, `layout`, `frames`:

- `frames[i].data` follows the same validation as `data` itself
- `frames[i].layout` follows the same validation as `layout`

I'm not 100% sure what that means for validation, but on a strictly abstract level, it's conceptually simple.